app/src/main/java/com/eveningoutpost/dexdrip/Models/Treatments.java

```diff
@@ -589,42 +633,22 @@ public static CobCalc cobCalc(Treatments treatment, double lastDecayedBy, double
         }
     }
 
-    public static Iob calcTreatment(Treatments treatment, double time, double lastDecayedby) {
-
-        final double dia = Profile.insulinActionTime(time); // duration insulin action in hours
-        final double peak = 75; // minutes in based on a 3 hour DIA - scaled proportionally (orig 75)
-        //final double sens = Profile.getSensitivity(time); // sensitivity currently in mmol
-        double insulin_delay_minutes = 0;
-
-        double insulin_timestamp = treatment.timestamp + (insulin_delay_minutes * 60 * 1000);
+    public static Iob calcTreatment(Treatments treatment, double time) {
 
         Iob response = new Iob();
-
-        final double scaleFactor = 3.0 / dia;
         double iobContrib = 0;
-        //double activityContrib = 0;
-
-        // only use treatments with insulin component which have already happened
-        if ((treatment.insulin > 0) && (insulin_timestamp < time)) {
-            //  double bolusTime = insulin_timestamp; // bit of a dupe
-            double minAgo = scaleFactor * (((time - insulin_timestamp) / 1000) / 60);
-
-            if (minAgo < peak) {
-                double x1 = minAgo / 5 + 1;
-                iobContrib = treatment.insulin * (1 - 0.001852 * x1 * x1 + 0.001852 * x1);
-                // units: BG (mg/dL)  = (BG/U) *    U insulin     * scalar
-                // activityContrib = sens * activityMultipler * treatment.insulin * (2 / dia / 60 / peak) * minAgo;
-
-            } else if (minAgo < 180) {
-                double x2 = (minAgo - peak) / 5;
-                iobContrib = treatment.insulin * (0.001323 * x2 * x2 - .054233 * x2 + .55556);
-                //   activityContrib = sens * activityMultipler * treatment.insulin * (2 / dia / 60 - (minAgo - peak) * 2 / dia / 60 / (60 * dia - peak));
-            }
+        double activityContrib = 0;
```

Comment:
How does this model compare numerically to the existing one which uses `Profile.insulinActionTime()` ?

Reply:
i analyzed your code to calculate IOB and activity and found a more generic calculation scheme by using "linear trapezoid curves"  as described in issue #718 (20th and 25th of august this year). So i compared your curve with mine and they are matching.
I only added a performance optimization (as answered to @k2s a week ago) to break prediction after 6 hours in future - we can change this behaviour if needed to a more flexible logic

Reply:
* I created #1054 because of the optimization vs. Lantus curve as I have not enough knowledge about the problem
* @gruoner pls. don't forget my [pull request](https://github.com/gruoner/xDrip/pull/20), curent state will crash xDrip in some situations